testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/EEConcurrencyExecutorShutdownTestCase.java

```diff
@@ -0,0 +1,230 @@
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeoutException;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.*;
+import static org.jboss.as.controller.operations.common.Util.createAddOperation;
+import static org.jboss.as.controller.operations.common.Util.createEmptyOperation;
+import static org.jboss.as.test.integration.domain.BuildConfigurationTestBase.createConfiguration;
+import static org.jboss.as.test.integration.domain.management.util.DomainTestSupport.cleanFile;
+import static org.jboss.as.test.integration.domain.management.util.DomainTestUtils.executeForResult;
+import static org.jboss.as.test.integration.management.util.ModelUtil.createOpNode;
+import static org.junit.Assert.fail;
+
+/**
+ * The test case schedules an instance of task ReSchedulingTask.
+ * Each instance of ReSchedulingTask sleeps for 10 seconds and then re-schedules another instance of its own class.
+ * After the CLI command /host=master/server-config=server-one:stop() is invoked the server should stop.
+ * Test for [ WFCORE-3868 ].
+ *
+ * @author Daniel Cihak
+ */
+public class EEConcurrencyExecutorShutdownTestCase {
+
+    private static final String ARCHIVE_FILE_NAME = "test.war";
+    private static final PathAddress ROOT_DEPLOYMENT_ADDRESS = PathAddress.pathAddress(DEPLOYMENT, ARCHIVE_FILE_NAME);
+    private static final PathAddress MAIN_SERVER_GROUP_ADDRESS = PathAddress.pathAddress(SERVER_GROUP, "main-server-group");
+    private static final PathAddress MAIN_SERVER_GROUP_DEPLOYMENT_ADDRESS = MAIN_SERVER_GROUP_ADDRESS.append(DEPLOYMENT, ARCHIVE_FILE_NAME);
+    private static final String FIRST_SERVER_NAME = "main-one";
+
+    private static DomainTestSupport testSupport;
+    private static DomainLifecycleUtil domainMasterLifecycleUtil;
+    private static File tmpDir;
+
+    private static DomainClient masterClient;
+
+    @BeforeClass
+    public static void setupDomain() {
+        testSupport = DomainTestSuite.createSupport(EEConcurrencyExecutorShutdownTestCase.class.getSimpleName());
+        domainMasterLifecycleUtil = testSupport.getDomainMasterLifecycleUtil();
+        masterClient = domainMasterLifecycleUtil.getDomainClient();
+    }
+
+    @Before
+    public void createDeployment() throws Exception {
+        WebArchive testDeployment = ShrinkWrap.create(WebArchive.class, ARCHIVE_FILE_NAME);
+        testDeployment.addClasses(TaskSchedulerServletContextListener.class);
+        tmpDir = new File("target/deployments/" + this.getClass().getSimpleName());
+        new File(tmpDir, "archives").mkdirs();
+        testDeployment.as(ZipExporter.class).exportTo(new File(tmpDir, "archives/" + ARCHIVE_FILE_NAME), true);
+
+        final WildFlyManagedConfiguration config = createConfiguration("domain.xml", "host.xml", getClass().getSimpleName());
+        config.setAdminOnly(true);
```

Comment:
At this point, the domain was already created and started. WildFlyManagedConfiguration object creation and configuration here looks like does nothing and can be removed.